experimental/harfbuzz/src/lib.rs

```diff
@@ -315,124 +154,521 @@ impl UnicodeFuncs {
     pub fn as_ptr(&self) -> *mut hb_unicode_funcs_t {
         self.raw
     }
-}
 
-impl Drop for UnicodeFuncs {
-    fn drop(&mut self) {
+    fn empty() -> Result<Self, HarfBuzzError> {
+        let empty = unsafe { hb_unicode_funcs_get_empty() };
+        // The HarfBuzz refcounting convention is that "create"
+        // sets refcount to one, not zero.
+        // https://harfbuzz.github.io/object-model-lifecycle.html
+        let raw = unsafe { hb_unicode_funcs_create(empty) };
+        if raw == empty {
+            Err(HarfBuzzError::Alloc)
+        } else {
+            Ok(Self { raw })
+        }
+    }
+
+    /// Sets up a `hb_unicode_funcs_t` with ICU4X compiled data as the back end as the Unicode
+    /// Database operations that HarfBuzz needs. The `hb_unicode_funcs_t` held
+    /// by the returned `UnicodeFuncs` is marked immutable.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    pub fn new() -> Result<Self, HarfBuzzError> {
+        let ufuncs = Self::empty()?;
+
         unsafe {
-            hb_unicode_funcs_destroy(self.raw);
+            hb_unicode_funcs_set_combining_class_func(
+                ufuncs.raw,
+                Some({
+                    extern "C" fn cb(
```

Comment:
This works, with the Rust fns all named `cb`? I guess the names get mangled? Nice